src/ipc-frontend-dbus.c

```diff
@@ -607,6 +609,7 @@ on_name_acquired (GDBusConnection *connection,
         &error);
     if (ret == FALSE)
         g_warning ("failed to export interface: %s", error->message);
+    self->connection_state = CONNECTION_STATE_CONNECTED;
```

Comment:
I think this state is misleading. Is not about establishing or not a connection, but about obtaining or not a well known name.

If a connection can't be made to the bus, then the on_name_lost callback handler will be called with a NULL GDBusConnection (and the bus_acquired_handler callback will never be called).

So if this is about making sure that a connection was made, then this has to be set in on_bus_acquired. But the intention is to know if the on_name_acquired callback was really called (since if the name can't be acquired, then on_name_acquired will never be called, just bus_acquired_handler followed by name_lost_handler).

That why I would instead use something like BUS_NAME_ACQUIRED. In fact, I don't think you need an enum, just a bool name_acquired should be enough.

Reply:
This is a good simplification. Will give it a try to see how it looks in practice.